auslib/db.py

```diff
@@ -1764,6 +1765,56 @@ def _putWatchdogValue(self, changed_by, value, where=None, transaction=None):
             super(Dockerflow, self).update(where=where, what=value, changed_by=changed_by, transaction=transaction)
 
 
+class Shutoffs(AUSTable):
+    def __init__(self, db, metadata, dialect):
+        self.table = Table('shutoffs', metadata,
+                           Column('shutoff_id', Integer, primary_key=True,
+                                  autoincrement=True),
+                           Column('product', String(15), nullable=False),
+                           Column('channel', String(75)),
+                           Column('mapping', String(100)),
+                           UniqueConstraint('product', 'channel')
+                           )
+        AUSTable.__init__(self, db, dialect, history=False, versioned=False)
+
+    def getShutoff(self, transaction=None, **columns):
+        """Returns a tuple where the first field says if the product+channel
+        combo is shutoff or not, and the second field specifies the mapped
+        empty release"""
+        # Since we are able to shutoff entire products without specifying a
+        # channel, we first get all shutoffs for a product and check if any of
+        # them has 'channel' set to None
+        matched_shutoffs = self.select(where=[self.product == columns.get('product')],
+                                       transaction=transaction)
+        for shutoff in matched_shutoffs:
+            if shutoff.get('channel') is None or shutoff.get('channel') == columns.get('channel'):
+                return (True, shutoff.get('mapping'))
+        return (False, None)
```

Comment:
If mapping ends up being NOT NULL, you can simplify this interface and just return the blob name, or None if there's no shutoff.